File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    
    dic =  {}
    for i in participant:
        if i in dic:
            dic[i] += 1
        else:
            dic[i] = 1
        
        
    for i in completion:
        dic[i] -= 1
        if dic[i] == 0:
            del dic[i]
    
    answer = list(dic.keys())[0]
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> map = new HashMap<>();
        
        for(int i=0; i<participant.length; i++){
            map.put(participant[i], map.getOrDefault(participant[i],0) + 1);
        }
        
        for(int i=0; i<completion.length; i++){
            map.put(completion[i], map.getOrDefault(completion[i],0) - 1);
        }
        
        List<String> keys = new ArrayList<>(map.keySet());
        
        keys.sort((o1,o2) -> map.get(o2) - map.get(o1));
        
        return keys.get(0);
    }
}